components/layout/layout.tsx

```diff
@@ -38,13 +38,9 @@ function generator({ suffixCls, tagName }: GeneratorProps) {
 
 class Basic extends React.Component<BasicProps, any> {
   render() {
-    const { prefixCls, className, children, tagName: CustomElement, ...others } = this.props;
+    const { prefixCls, className, children, tagName, ...others } = this.props;
     const classString = classNames(className, prefixCls);
-    return (
-      <CustomElement className={classString} {...others}>
-        {children}
-      </CustomElement>
-    );
+    return React.createElement(tagName!, { className: classString, ...others }, children);
```

Comment:
Still not a perfect fix, can we avoid non-null assertion?

Reply:
I think it is safe enough to do that assertion. The end-users either can provide a valid `tagName` or don't provide anything.

Reply:
Yes, it does not affect end-users, but it's unsafe to the antd itself codebase.

I'm ok for this quick fix.  But it would be better if we could not use the non-null assertion.